modules/tide_site_simple_sitemap/src/Plugin/simple_sitemap/SitemapGenerator/TideDefaultSitemapGenerator.php

```diff
@@ -89,22 +93,21 @@ public function generate(array $links) {
             $links_per_site[$site_id][] = $site_link;
           }
         }
-        \Drupal::service('entity.memory_cache')->deleteAll();
       }
 
       // Now we have the sitemap of all sites.
       foreach ($links_per_site as $site_id => $site_links) {
-        if (count($site_links)) {
+        if (!empty($site_links)) {
           // Write to our own table.
           $values = [
-            'id' => NULL === $highest_id ? 0 : $highest_id + 1,
+            'id' => NULL === $highest_id ? 0 : $highest_id,
```

Comment:
The principle of this mofule is to serve site-based sitemap. Whatever sitemap generated by the contrib module, tide site map will split it into sites. The ID of the site-based sitemaps must be the same with the parent sitemap.

The parent sitemap has ID as the primary key. The site-based sitemaps use (ID, Site ID) as the key.

Reply:
thanks @sonnykt . changed.